### PR TITLE
`[EXILED::API]` Adding SpawnMice

### PR DESCRIPTION
**Pull request by NotZer0Two**
Spawns rats.

(What to you expect something incredible nope it just spawns rats)

### CODE REVIEW

**Comment by NotZer0Two** [comment]
Credits goes to yamato for the original finding

**Comment by NotZer0Two** [comment]
Im fixing the issue and i will post a bug fix

**Comment by NotZer0Two** [comment]
The issues where that Mouse spawn could overflow and cause crash and optimized the method because was resource consuming

File: Exiled.API/Features/Map.cs [hunk 1]
```diff
@@ -353,6 +359,19 @@ public static void PlayGunSound(Vector3 position, ItemType firearmType, byte max
             msg.SendToAuthenticated();
         }
 
+        /// <summary>
+        /// Spawns a Mice inside the <see cref="RoomType.EzShelter"/>.
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Map.cs:363]:
```suggestion
        /// Spawns mice inside the <see cref="RoomType.EzShelter"/>.
```

File: Exiled.API/Features/Map.cs [hunk 2]
```diff
@@ -353,6 +359,19 @@ public static void PlayGunSound(Vector3 position, ItemType firearmType, byte max
...
+        /// </summary>
+        /// <param name="mice">The type of mice you want to spawn..</param>
+        public static void SpawnMice(byte mice = 1)
+        {
+            if (mice > SqueakSpawner.mice.Length)
+                new OverflowException($"The mice type are from 1 to {SqueakSpawner.mice.Length}");
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Map.cs:369]:
```suggestion
                throw new OverflowException($"Mouse type must be between 1 and {SqueakSpawner.mice.Length}.");
```

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

**Review by xNexusACS — APPROVED** [review]

**Pull request merged**